lib/postprocessing/image.js

```diff
@@ -265,7 +272,7 @@ async function needsImagePostProcess(rendition, source) {
     }
 
     // check if it can read the intermediate rendition
-    if (!isSupportedInputFormat(metadata)) {
+    if (!isSupportedInputFormat(metadata.format)) {
         console.log(`skipping post-processing because intermediate rendition format '${metadata.format}' is not supported`);
```

Comment:
Good question, not sure myself.

Facts:
- We get here if the output format in `fmt` would be supported (png, jpg, gif, tiff, webp) but the intermediate output format (as seen by imagemagick content type detection) is not something known.
  => I noticed there might be a bug in that the `rendition` passed into `needsImagePostProcess()` would not have the original instructions if `prepareImagePostProcess()` changed them (as in the webp case). This might affect a few things...
- It would only end up here if a worker code is faulty - generate an intermediate rendition that is not confirming to the requested `fmt`. The requested `fmt` for the intermediate rendition is now ensured by the new `prepareImagePostProcess()` (new in NUI-867)
- The logic to skip has been in [since post-processing was added](https://github.com/adobe/asset-compute-sdk/commit/ffb5248ffbd1d847ddfeafe88068f20d097f112a) and has been working fine in all tests and production so far.
- If it throws, it would fail the processing.
- If it skips (as currently), it would just pass through the result rendition of the worker as the final result.
- Changing it to throw doesn't break the tests of sdk. (Haven't checked dependent_module_tests).

Reply:
Yeah. I am trying out the change: https://github.com/adobe/asset-compute-sdk/pull/66/commits/cafe011e8331847874e1a3be1dc5d082636f019e

Let's see if it has any impact on workers in dependent module or IT tests.